### ISSUE TITLE
compilation error: undeclared (first use in this function)..

### ISSUE THREAD

**Issue by e1voronov** [bootlin/libva-v4l2-request#7]
Could you please help to resolve compilation issue. Thank you!

>   CC       v4l2_request_drv_video_la-config.lo
config.c: In function ‘RequestQueryConfigProfiles’:
config.c:116:6: error: ‘V4L2_PIX_FMT_MPEG2_SLICE’ undeclared (first use in this function); did you mean ‘V4L2_PIX_FMT_MPEG2’?
      V4L2_PIX_FMT_MPEG2_SLICE);
      ^~~~~~~~~~~~~~~~~~~~~~~~
      V4L2_PIX_FMT_MPEG2
config.c:116:6: note: each undeclared identifier is reported only once for each function it appears in
config.c:124:6: error: ‘V4L2_PIX_FMT_H264_SLICE’ undeclared (first use in this function); did you mean ‘V4L2_PIX_FMT_H264_MVC’?
      V4L2_PIX_FMT_H264_SLICE);
      ^~~~~~~~~~~~~~~~~~~~~~~
      V4L2_PIX_FMT_H264_MVC
config.c:135:6: error: ‘V4L2_PIX_FMT_HEVC_SLICE’ undeclared (first use in this function); did you mean ‘V4L2_PIX_FMT_H264_SLICE’?
      V4L2_PIX_FMT_HEVC_SLICE);
      ^~~~~~~~~~~~~~~~~~~~~~~
      V4L2_PIX_FMT_H264_SLICE
Makefile:561: recipe for target 'v4l2_request_drv_video_la-config.lo' failed
make[2]: *** [v4l2_request_drv_video_la-config.lo] Error 1
make[2]: Leaving directory '/usr/src/libva-v4l2-request/src'
Makefile:393: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/usr/src/libva-v4l2-request/src'
Makefile:393: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

**Comment by TarkusGSA** [comment]
I have absolutely the same issue.

**Comment by Dreyk007** [comment]
Same problem for me. But i have linux-cedrus headers installed. How else can I fix this error?

**Comment by maximst** [comment]
Yes, I am have linux-cedrus headers installed to. But get the same error.

**Comment by adriancable** [comment]
Same problem for me - and I do have the headers installed. Did we ever find a resolution for this?

**Comment by luusl** [comment]
This repository hasn't been updated for the latest kernel sources. I'm using on of pH5's branches: https://github.com/pH5/libva-v4l2-request/tree/multi-context

**Comment by roberthoegerl** [comment]
> This repository hasn't been updated for the latest kernel sources. I'm using on of pH5's branches: https://github.com/pH5/libva-v4l2-request/tree/multi-context

Tried both pH5 and bootlin, but the problem stays the same: 

```
config.c: In function 'RequestQueryConfigProfiles':
config.c:123:6: error: 'V4L2_PIX_FMT_MPEG2_SLICE' undeclared (first use in this function); did you mean 'V4L2_PIX_FMT_HEVC_SLICE'?
      V4L2_PIX_FMT_MPEG2_SLICE);
      ^~~~~~~~~~~~~~~~~~~~~~~~
      V4L2_PIX_FMT_HEVC_SLICE
config.c:123:6: note: each undeclared identifier is reported only once for each function it appears in
Makefile:601: recipe for target 'v4l2_request_drv_video_la-config.lo' failed
```

> You've very likely not installed the headers of our kernel, which are adding those definitions.

I am using odroid's ubuntu kernel (linux-image-xu3 _aka_ linux-image-4.14.111-139) for which there is no ready kernel header ubuntu package, so I cloned the headers using git:
`git clone --depth 1 https://github.com/hardkernel/linux -b odroidxu4-4.14.y`
and created links
```
ln -s /root/linux /lib/modules/$(uname -r)/build/include/linux
ln -s /root/linux /usr/src/linux-headers-$(uname -r)/include/linux
```
Still getting the same error.
